filebeat/input/filestream/input.go

```diff
@@ -264,7 +264,7 @@ func (inp *filestream) openFile(path string, offset int64) (*os.File, error) {
 	return f, nil
 }
 
-func (inp *filestream) checkFileBeforeOpening(path string) error {
+func (inp *filestream) checkFileBeforeOpening(path string, offset *int64) error {
```

Comment:
nits:
- if possible do not modifiy the offset via pointer, but return the updated offset as a value.
- we need this to be a method on `*filestream`?

Maybe we want to split the validation from computing the initial offset. Especially regarding that we might want to introduce settings that can start from a different offset.